src/main/java/billgates/view/gui/SignInDialog.java

```diff
@@ -0,0 +1,110 @@
+            if (true){
+                JPanel panel = new JPanel();
+                SpringLayout layout = new SpringLayout();
+                panel.setLayout(layout);
+                root.add(panel, BorderLayout.CENTER);
+
+                JLabel nameLabel = new JLabel("Name:");
+                JLabel passwordLabel = new JLabel("Password:");
+                panel.add(nameLabel);
+                panel.add(nameInput);
+                panel.add(passwordLabel);
+                panel.add(passwordInput);
+
+                SpringLayout.Constraints nLCons = layout.getConstraints(nameLabel);
+                nLCons.setX(Spring.constant(52));
+                nLCons.setY(Spring.constant(30));
+                nLCons.setWidth(Spring.constant(38));
+                nLCons.setHeight(Spring.constant(20));
+
+                SpringLayout.Constraints pLCons = layout.getConstraints(passwordLabel);
+                pLCons.setWidth(Spring.constant(60));
+                pLCons.setHeight(Spring.constant(20));
+                pLCons.setX(Spring.constant(30));
+                pLCons.setY(Spring.sum(Spring.constant(50), nLCons.getHeight()));
+
+                SpringLayout.Constraints nICons = layout.getConstraints(nameInput);
+                nICons.setX(Spring.sum(Spring.sum(pLCons.getX(), pLCons.getWidth()), Spring.constant(5)));
+                nICons.setY(nLCons.getY());
+
+                SpringLayout.Constraints pICons = layout.getConstraints(passwordInput);
+                pICons.setX(nICons.getX());
+                pICons.setY(pLCons.getY());
+
+                passwordInput.setEchoChar('*');
+            }
+
+            if (true){
+                JPanel panel = new JPanel();
+                root.add(panel, BorderLayout.SOUTH);
+
+                panel.add(registerButton);
+                panel.add(signOnButton);
+            }
+
+            // Case 1: Click registerButton
+            registerButton.addActionListener(e -> {
+                setVisible(false);
+                RegisterDialog dlg = new RegisterDialog(this);
+                if (dlg.exec()){
+                    JOptionPane.showMessageDialog(null,
+                            "Account has been successfully created !", "Congratulation",
+                            JOptionPane.PLAIN_MESSAGE);
+                    System.out.println(dlg.getInput());
+                }
+            });
+
+            // Case 2: Click signOnButton
+            signOnButton.addActionListener((e -> {
+                accepted = true;
+                setVisible(false);
+            }));
+        }
+
+        public String getInput(){
+            return String.format("Name: %s\n Password: %s", nameInput.getText(),
+                    Arrays.toString(passwordInput.getPassword()));
+        }
+
+        // Execute
+        public boolean exec(){
+            this.setTitle("Sign In");
+            this.setModal(true);
+            this.setSize(300, 200);
+            this.setResizable(false);
```

Comment:
![image](https://user-images.githubusercontent.com/8016180/201283487-47432435-8ffc-47c0-91de-de807eb12a9f.png)
I'm also not sure if this is the intended display, it seems like some of the words are not being displayed properly? (Perhaps of the Swing constants? I'm not sure what they are doing)

Reply:
For windows,
![image](https://user-images.githubusercontent.com/60980117/201430783-97e8b303-a158-4328-8e32-4bfc352aff94.png)